### ISSUE TITLE
SQLiteException: no such collation sequence

### ISSUE THREAD

**Issue by vanniktech** [couchbase/couchbase-lite-android#987]
When initializing the database using

```java
final Manager manager = new Manager(new AndroidContext(context), Manager.DEFAULT_OPTIONS);
database = manager.getDatabase(DatabaseManager.DATABASE_NAME);
```

I get an exception:

> com.couchbase.lite.storage.SQLException: android.database.sqlite.SQLiteException: no such collation sequence: REVID (code 1): , while compiling: CREATE TABLE revs (         sequence INTEGER PRIMARY KEY AUTOINCREMENT,         doc_id INTEGER NOT NULL REFERENCES docs(doc_id) ON DELETE CASCADE,         revid TEXT NOT NULL COLLATE REVID,         parent INTEGER REFERENCES revs(sequence) ON DELETE SET NULL,         current BOOLEAN,         deleted BOOLEAN DEFAULT 0,         json BLOB,         UNIQUE (doc_id, revid))

When the manager will run this query:

```sql
CREATE TABLE revs (         sequence INTEGER PRIMARY KEY AUTOINCREMENT,         doc_id INTEGER NOT NULL REFERENCES docs(doc_id) ON DELETE CASCADE,         revid TEXT NOT NULL COLLATE REVID,         parent INTEGER REFERENCES revs(sequence) ON DELETE SET NULL,         current BOOLEAN,         deleted BOOLEAN DEFAULT 0,         json BLOB,         UNIQUE (doc_id, revid))
```

as part of the open method when initializing the schema on a db version of 0.

I'm using version 1.1.0 of the library and this crash seems to happen only on Android N. Upgrading to the latest version is not really an option since the project I'm involved with used private parts of the library which are no longer present in version 1.3.0

**Comment by hideki** [comment]
Hi @vanniktech,

CBL v1.2.x or earlier can not support Android N because Android N does not allow the application to use most of native libraries that are provided by Android. 

Thanks,
Hideki

**Comment by vanniktech** [comment]
Thanks for your answer Hideki :)
So the only way is upgrading to 1.3? Also since I just got into that project and never used the couchbase library before, can the library handle an upgrade from one version to another (e.g. 1.1.0 to 1.3.0) without loosing the persisted data from the 1.1.0 time?

**Comment by hideki** [comment]
Hi @vanniktech,
CBL provides auto database upgrading functionality. With 1.2.x and 1.3.0, CBL Java has database upgrading issue. See #888 . This issue is addressed with 1.3.1 release which will be released soon. I recommend you to wait till 1.3.1 is released or make a build from source code. 
Thanks!
